examples/google_cloud_functions/oauth_main.py

```diff
@@ -0,0 +1,101 @@
+# https://cloud.google.com/functions/docs/first-python
+
+import logging
+
+from slack_bolt.oauth.oauth_settings import OAuthSettings
+
+logging.basicConfig(level=logging.DEBUG)
+
+from slack_bolt import App
+from datastore import GoogleDatastoreInstallationStore, GoogleDatastoreOAuthStateStore
+
+from google.cloud import datastore
+
+datastore_client = datastore.Client()
+logger = logging.getLogger(__name__)
+
+# process_before_response must be True when running on FaaS
+app = App(
+    process_before_response=True,
+    installation_store=GoogleDatastoreInstallationStore(
+        datastore_client=datastore_client,
+        logger=logger,
+    ),
+    oauth_settings=OAuthSettings(
+        state_store=GoogleDatastoreOAuthStateStore(
+            datastore_client=datastore_client,
+            logger=logger,
+        ),
+    ),
+)
+
+
+@app.command("/hello-bolt-python-gcp")
+def hello_command(ack):
+    ack("Hi from Google Cloud Functions!")
+
+
+@app.event("app_mention")
+def event_test(body, say, logger):
+    logger.info(body)
+    say("Hi from Google Cloud Functions!")
+
+
+# Flask adapter
+# TODO: Once this once a new version newer than 1.13.2, delete handler and enable the slack_bolt.adapter import instead
+# from slack_bolt.adapter.google_cloud_functions import SlackRequestHandler
+from handler import SlackRequestHandler
+from flask import Request
+
+
+handler = SlackRequestHandler(app)
+
+
+# Cloud Function
+def hello_bolt_app(req: Request):
+    """HTTP Cloud Function.
+    Args:
+        req (flask.Request): The request object.
+        <https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data>
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <https://flask.palletsprojects.com/en/1.1.x/api/#flask.make_response>.
+    """
+    return handler.handle(req)
+
+
+# For local development
+# python main.py
+if __name__ == "__main__":
+    from flask import Flask, request
+
+    flask_app = Flask(__name__)
+
+    @flask_app.route("/function", methods=["GET", "POST"])
+    def handle_anything():
+        return handler.handle(request)
+
+    flask_app.run(port=3000)
+
+
+# Step1: Create a new Slack App: https://api.slack.com/apps
+# Bot Token Scopes: chat:write, commands, app_mentions:read
```

Comment:
FYI the .env.yaml.oauth-sample has a different list than this; it includes `channels.history`.

Reply:
Thanks! I've updated the part.